tests/http/httpClient/Middleware/BodyInspectionHandlerTests.cs

```diff
@@ -0,0 +1,112 @@
+using System.Net.Http;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+using Microsoft.Kiota.Http.HttpClientLibrary.Tests.Mocks;
+using Xunit;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Tests.Middleware;
+
+public class BodyInspectionHandlerTests : IDisposable
+{
+    private readonly List<IDisposable> _disposables = [];
+
+    [Fact]
+    public void BodyInspectionHandlerConstruction()
+    {
+        using var defaultValue = new BodyInspectionHandler();
+        Assert.NotNull(defaultValue);
+    }
+
+    [Fact]
+    public async Task BodyInspectionHandlerGetsRequestBodyStream()
+    {
+        var option = new BodyInspectionHandlerOption { InspectRequestBody = true, };
+        using var invoker = GetMessageInvoker(new HttpResponseMessage(), option);
+
+        // When
+        var request = new HttpRequestMessage(HttpMethod.Post, "https://localhost")
+        {
+            Content = new StringContent("request test")
+        };
+        var response = await invoker.SendAsync(request, default);
+
+        // Then
+        Assert.NotNull(option.RequestBody);
+        Assert.Equal("request test", GetStringFromStream(option.RequestBody!));
+    }
```

Comment:
Since the test invoker does not read the http content(its not really sending it) should we add an assertion that the content is still intact/readable/valid?

Reply:
Makes sense, I added an assert, similar to how I did it for response assertion.